### ISSUE TITLE
Update Font Awesome to v6

### ISSUE THREAD

**Issue by platschi** [siteorigin/vantage#392]
Vantage is using an outdated version of Font Awesome (4.7) which misses out  on thousands of new icons, among them covid-19 related icons and so on.

It would be awesome if it's possible to update to v5 or even better give theme users the option to select between v4.7 or v5.15.3 (current) for compatibility purposes.

**Comment by Misplon** [comment]
Hi @platschi, thanks for reaching out. 

The Widgets Bundle is up to date. If you are using the Circle Icon widget you can instead switch to the Features widget. If the icons are required for menu icons then another solution will be needed. You can perhaps install the Font Awesome plugin: https://wordpress.org/plugins/font-awesome/. We could assist with any custom CSS required for menu icons.

Vantage is using the last version of Font Awesome version 4. The upgrade to version 5 will require some migration.

**Comment by Misplon** [comment]
Hi @platschi 

I had a quick look at our work timeline. It'll probably be a couple of months before we can implement the upgrade. We're completing and starting a few new widgets for the Widgets Bundle that have been in the queue for some time. If there is anything we do to assist with your immediate requirements, please, let us know.

Thanks again for logging this.

Cheers, Andrew

**Comment by Misplon** [comment]
Retargeting this to Font Awesome 6.

Version 5 was current when this was logged. It is now two major versions behind, so upgrading to 5 would mean repeating the work later. Version 6 keeps the class names introduced in 5 and adds the larger icon set that was asked for here.

The upgrade path already exists in the theme. Vantage moved from Font Awesome 3.2.1 to 4 using a rename map in `fontawesome/icon-migration.php`, applied through the `vantage_fontawesome_icon_name` filter. Menu icons and the Circle Icon widget both pass through that filter, so saved icon names on existing sites can be mapped forward the same way.

One extra step is needed this time. Version 5 split the free set into solid, regular and brands, each with its own prefix, so `fa fa-star` becomes `fas fa-star`. The filter returns an icon name only, so it needs to return the style prefix as well.

Scope:

* Replace the font assets and stylesheet in `fontawesome/`.
* Add a 4 to 6 rename map alongside the existing 3 to 4 map.
* Extend `vantage_fontawesome_icon_name` to carry the style prefix.
* Rebuild the icon picker data in `fontawesome/icons.php` and `fontawesome/icon-sections.php`.

**Comment by Misplon** [comment]
Done in develop.

Font Awesome is now 6.7.2. The version 4 shim ships alongside it, so icons saved before the upgrade still draw. Of the 766 icons the theme offered, 625 draw natively in version 6 and 139 draw through the shim.

The menu icon and Circle Icon pickers now list 2024 icons, grouped by the 68 categories Font Awesome publishes. They listed 764 before.

Font Awesome removed meanpath and tripadvisor. Saved values for those fall back to search and suitcase.

48 version 4 names that version 6 folded into another icon are mapped to the surviving name. Every pair was checked in a browser to confirm the drawn glyph, font family and weight stay the same.
